### ISSUE TITLE
 eslint crash when upgrading to 1.4.1

### ISSUE THREAD

**Issue by bloody-ux** [estools/esquery#135]
When we upgrade esquery to 1.4.1, eslint will crash, below is the log：

> eslint ./src --ext .jsx,.js,.ts,.tsx

Oops! Something went wrong! :(

ESLint: 7.32.0

TypeError: Cannot read properties of undefined (reading 'type')
Occurred while linting /Users/bhtbed/workspace/特价版/merge-radish/src/app.ts:1
    at /Users/bhtbed/workspace/特价版/merge-radish/node_modules/_esquery@1.4.1@esquery/dist/esquery.min.js:1:29531
    at /Users/bhtbed/workspace/特价版/merge-radish/node_modules/_esquery@1.4.1@esquery/dist/esquery.min.js:1:30574
    at Function.b.matches (/Users/bhtbed/workspace/特价版/merge-radish/node_modules/_esquery@1.4.1@esquery/dist/esquery.min.js:1:34910)
    at NodeEventGenerator.applySelector (/Users/bhtbed/workspace/特价版/merge-radish/node_modules/_eslint@7.32.0@eslint/lib/linter/node-event-generator.js:292:21)
    at NodeEventGenerator.applySelectors (/Users/bhtbed/workspace/特价版/merge-radish/node_modules/_eslint@7.32.0@eslint/lib/linter/node-event-generator.js:320:22)
    at NodeEventGenerator.enterNode (/Users/bhtbed/workspace/特价版/merge-radish/node_modules/_eslint@7.32.0@eslint/lib/linter/node-event-generator.js:336:14)
    at CodePathAnalyzer.enterNode (/Users/bhtbed/workspace/特价版/merge-radish/node_modules/_eslint@7.32.0@eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
    at /Users/bhtbed/workspace/特价版/merge-radish/node_modules/_eslint@7.32.0@eslint/lib/linter/linter.js:960:32
    at Array.forEach (<anonymous>)

**Comment by binsee** [comment]
same problem

- output
```
> file-box@1.7.0 lint:es
> eslint --ignore-pattern fixtures/ "src/**/*.ts" "tests/**/*.ts"

Oops! Something went wrong! :(

ESLint: 8.34.0

TypeError: Cannot read properties of undefined (reading 'type')
Occurred while linting /Users/binsee/code/file-box/src/config.ts:1
    at /Users/binsee/code/file-box/node_modules/esquery/dist/esquery.min.js:1:29496
    at /Users/binsee/code/file-box/node_modules/esquery/dist/esquery.min.js:1:30539
    at Function.b.matches (/Users/binsee/code/file-box/node_modules/esquery/dist/esquery.min.js:1:34875)
    at NodeEventGenerator.applySelector (/Users/binsee/code/file-box/node_modules/eslint/lib/linter/node-event-generator.js:296:21)
    at NodeEventGenerator.applySelectors (/Users/binsee/code/file-box/node_modules/eslint/lib/linter/node-event-generator.js:324:22)
    at NodeEventGenerator.enterNode (/Users/binsee/code/file-box/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/Users/binsee/code/file-box/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
    at /Users/binsee/code/file-box/node_modules/eslint/lib/linter/linter.js:1153:32
    at Array.forEach (<anonymous>)
    at runRules (/Users/binsee/code/file-box/node_modules/eslint/lib/linter/linter.js:1148:15)
```


- package-lock.json

```
    "node_modules/esquery": {
      "version": "1.4.1",
      "resolved": "https://registry.npmmirror.com/esquery/-/esquery-1.4.1.tgz",
      "integrity": "sha512-3ZggxvMv5EEY1ssUVyHSVt0oPreyBfbUi1XikJVfjFiBeBDLdrb0IWoDiEwqT/2sUQi0TGaWtFhOGDD8RTpXgQ==",
      "dev": true,
      "dependencies": {
        "estraverse": "^5.1.0"
      },
      "engines": {
        "node": ">=0.10"
      }
    },
```

**Comment by bmnidhin** [comment]
We are facing this too

**Comment by Silvea12** [comment]
Got it affecting a number of modules, including the `@typescript-eslint/require-await` rule from https://github.com/typescript-eslint/typescript-eslint, as seen in https://github.com/typescript-eslint/typescript-eslint/issues/6477

**Comment by markchristianlacap** [comment]
same problem with my eslint config

**Comment by bradzacher** [comment]
A mocha test case for the repo which should pass, but fails in the latest version

```js
    it("child not", function () {
        const program = {
            type: "Program",
            body: [
                {
                    type: "VariableDeclaration",
                    declarations: [
                        {
                            type: "VariableDeclarator",
                            id: { type: "Identifier", name: "x" },
                            init: { type: "Literal", value: 1, raw: "1" },
                        },
                    ],
                    kind: "var",
                },
            ],
        };
        const matches = esquery(program, "VariableDeclarator[init] > *");

        assert.includeMembers(matches, [
            program.body[0].declarations[0].id,
            program.body[0].declarations[0].init,
        ]);
    });
```

Running this test against the `v1.4.0` tag passes.
Bisecting I think this was introduced in 44e1ff540b1d3643e3d9f0ad53f490e9fc1888cc

**Comment by stheine** [comment]
Same issue here.
As workaround, on the top app level I install the older version:
`npm install --save-dev esquery@1.4.0`

**Comment by andreassjoberg** [comment]
> Same issue here. As workaround, on the top app level I install the older version: `npm install --save-dev esquery@1.4.0`

You can also add to your `package.json`:
```
  "overrides": {
    "esquery": "1.4.0"
  },
```

**Comment by sezanzeb** [comment]
In yarn, I think adding

```
    "resolutions": {
        "eslint/esquery": "1.4.0"
    },
```

to your package.json and running `yarn` would be a better solution than adding a fixed esquery version as a dependency of your project.

In npm I think you can use overrides: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

**Comment by tentone** [comment]
Same problem can be fixed by reverting back to version 1.4.0.

![bug](https://user-images.githubusercontent.com/9158354/219380056-ed3446e6-b2d5-4ad4-a4c3-b14cfdf17b98.PNG)

**Comment by michaelficarra** [comment]
This should be fixed by `1.4.2`. Please confirm.

**Comment by aarongerig** [comment]
@michaelficarra Upgrading to `1.4.2` solved the issue for me.

**Comment by davorpa** [comment]
> This should be fixed by `1.4.2`. Please confirm.

It works for me
